lib/archethic/p2p/message.ex

```diff
@@ -996,21 +996,26 @@ defmodule ArchEthic.P2P.Message do
     end
   end
 
-  def process(%GetTransactionChain{address: tx_address, after: date = %DateTime{}}) do
-    chain =
+  # current page state contains the page state that has already been with the node
+  def process(%GetTransactionChain{
+        address: tx_address,
+        after: after_time = %DateTime{},
+        page: current_page_state
+      }) do
+    {chain, new_page_state, more?} =
       tx_address
-      |> TransactionChain.get()
-      |> Stream.filter(&(DateTime.compare(&1.validation_stamp.timestamp, date) == :gt))
+      |> TransactionChain.get(after_time: after_time, page: current_page_state)
 
-    %TransactionList{
-      transactions: chain
-    }
+    # new_page_state contains the  page number of data being sent
+    %TransactionList{transactions: chain, page: new_page_state, more?: more?}
   end
 
-  def process(%GetTransactionChain{address: tx_address, after: nil}) do
-    %TransactionList{
-      transactions: TransactionChain.get(tx_address)
-    }
+  def process(%GetTransactionChain{address: tx_address, after: nil, page: current_page_state}) do
```

Comment:
Why do multiple calls ?
Is not the TransactionChain.get or the DB impl to manage the switch based on the options ?